src/main/java/com/facebook/openwifirrm/modules/ConfigManager.java

```diff
@@ -63,8 +66,27 @@ private class DeviceData {
 	/** Is the main thread sleeping? */
 	private final AtomicBoolean sleepingFlag = new AtomicBoolean(false);
 
-	/** Was a manual config update requested? */
-	private final AtomicBoolean eventFlag = new AtomicBoolean(false);
+	/**
+	 * Set of zones for which manual config updates have been requested.
+	 * <p>
+	 * This is a thread-safe (concurrent) set, since it is backed by a
+	 * {@code ConcurrentHashMap}, but is still statically typed as
+	 * {@code Set<String>} since there is nothing like ConcurrentHashSet in
+	 * Java.
+	 * <p>
+	 * This set is "concurrent but not synchronized" meaning it is thread-
+	 * safe, but simultaneous writes are allowed for different "hash buckets".
+	 */
+	private Set<String> zonesToUpdate = ConcurrentHashMap.newKeySet();
+
+	/**
+	 * This set is used as a buffer to track queued updates without actually
+	 * writing them to {@link #zonesToUpdate}. This is done in order to prevent
+	 * the possibility of trying to update zones in quick succession (which is
+	 * not allowed by the debounce timer).
+	 */
+	private Set<String> temporaryZonesToUpdateBuffer =
+		ConcurrentHashMap.newKeySet();
```

Comment:
I don't see the purpose of this - in what case does this actually help vs. using `zonesToUpdate` directly?